### PR TITLE
JSX support

### PR DESCRIPTION
**Pull request by ruscoder**
I've implemented support for JSX using acorn-jsx. Tests are included.

### CODE REVIEW

**Comment by BYK** [comment]
Hey, this looks awesome, thanks!

I'll check `acorn-jsx` and merge the PR if everything looks okay. Any reason not using Espree which supports all accepted ES6 features + JSX and is built on top of Acorn itself?

**Comment by jeresig** [comment]
I just confirmed, by running this in one of my projects, that this appears to work exactly as advertised! Thank you @ruscoder!

File: package.json [hunk 1]
```diff
@@ -43,11 +43,13 @@
...
   "dependencies": {
-    "acorn": "^2.7.0",
+    "acorn": "^3.0.0",
+    "acorn-jsx": "^3.0.0",
     "commander": "^2.9.0",
     "escape-string-regexp": "^1.0.4",
     "gettext-parser": "^1.1.2",
-    "jade": "^1.11.0"
+    "jade": "^1.11.0",
+    "object-assign": "^4.0.1"
```
Comment (BYK) [inline package.json]: I don't think it's worth adding a polyfill for something existing in Node 4. You also use this in a single place. Do you think we can remove this dependency?

**Comment by BYK** [comment]
@ruscoder - can you respond to my Espree question which is a single parser interface compared to the acorn + acorn-jsx solution in this patch?

Also you did some unrelated stylistic changes to the file. Can you revert them? Not that I like the existing style but I'd rather not change them to keep the diff as clean as possible. I plan to tidy up the styles by introducing ESLint at some point.

**Comment by ruscoder** [comment]
@BYK I haven't seen Espree before. Would you like to use Espree instead of acorn + acorn-jsx?

OK, I'll revert stylistic changes and notify you.

**Comment by BYK** [comment]
@ruscoder - if you didn't know about Espree it is okay to keep this as is. I mostly wanted to learn your reasoning behind choosing acorn-jsx over that assuming you knew about it :) 

Since this works along with tests I can always switch it to Espree in the future safely. So just revert the stylistic changes and I'll merge this away. 

Thanks!

**Comment by ruscoder** [comment]
@BYK I've updated pull request.

File: package.json (hunk 1 above)
Comment (ruscoder) [inline package.json]: You are using Node 0.10 and Node 0.11 in CI. This polyfill is required for now..

Comment (BYK) [inline package.json]: Oh, good point. I'll update these in the coming update.

**Comment by BYK** [comment]
Released 0.9.0

**Comment by jeresig** [comment]
Fantastic - thank you both! I've upgraded my app to `v0.9.0` and it works great.

**Pull request merged**